x/tariff/keeper/allocation.go

```diff
@@ -11,6 +11,9 @@ func (k Keeper) AllocateTokens(ctx sdk.Context) {
 
 	params := k.GetParams(ctx)
 	feesToDistribute := feesCollected.MulDecTruncate(params.Share)
+	if params.TransferFeeDenom == "" || feesToDistribute.AmountOf(params.TransferFeeDenom).IsZero() {
```

Comment:
```suggestion
	if feesToDistribute.AmountOf(params.TransferFeeDenom).IsZero() {
```

We still want to distribute fees to the distribution entities if the `TransferFeeDenom` is not set. `TransferFeeDenom` specifically applies to the BPS fee collection, but tariff will distribute all collected fees for all denoms.